src/protocol/ipa_prf/boolean_ops/comparison_and_subtraction_low_com.rs

```diff
@@ -0,0 +1,439 @@
+}
+
+#[cfg(all(test, unit_test))]
+mod test {
+    use rand::Rng;
+
+    use crate::{
+        ff::{
+            boolean::Boolean,
+            boolean_array::{BA32, BA64},
+            Expand, Field,
+        },
+        protocol,
+        protocol::{
+            context::Context,
+            ipa_prf::boolean_ops::comparison_and_subtraction_low_com::{
+                compare_geq, compare_gt, integer_sat_sub, integer_sub,
+            },
+        },
+        rand::thread_rng,
+        secret_sharing::{replicated::semi_honest::AdditiveShare, SharedValue},
+        test_executor::run,
+        test_fixture::{Reconstruct, Runner, TestWorld},
+    };
+
+    ///testing correctness of Not
+    /// just because we need it for subtractions
+    #[test]
+    fn test_not() {
+        assert_eq!(<Boolean>::ONE, !(<Boolean>::ZERO));
+        assert_eq!(<Boolean>::ZERO, !(<Boolean>::ONE));
+        assert_eq!(
+            AdditiveShare(<Boolean>::ZERO, <Boolean>::ZERO),
+            !AdditiveShare(<Boolean>::ONE, <Boolean>::ONE)
+        );
+        assert_eq!(
+            AdditiveShare(
+                <BA64>::expand(&<Boolean>::ZERO),
+                <BA64>::expand(&<Boolean>::ZERO)
+            ),
+            !AdditiveShare(
+                <BA64>::expand(&<Boolean>::ONE),
+                <BA64>::expand(&<Boolean>::ONE)
+            )
+        );
+        assert_eq!(
+            !AdditiveShare(
+                <BA64>::expand(&<Boolean>::ZERO),
+                <BA64>::expand(&<Boolean>::ZERO)
+            ),
+            AdditiveShare(
+                <BA64>::expand(&<Boolean>::ONE),
+                <BA64>::expand(&<Boolean>::ONE)
+            )
+        );
+    }
+
+    ///testing comparisons geq
+    #[test]
+    fn semi_honest_compare_geq() {
+        run(|| async move {
+            let world = TestWorld::default();
+
+            let mut rng = thread_rng();
+
+            let records: Vec<BA64> = vec![rng.gen::<BA64>(), rng.gen::<BA64>()];
+            let x = records[0].as_u128();
+            let y = records[1].as_u128();
+
+            let expected = x >= y;
+
+            let result = world
+                .semi_honest(records.clone().into_iter(), |ctx, x_y| async move {
+                    compare_geq::<_, BA64, BA64>(
+                        ctx.set_total_records(1),
+                        protocol::RecordId(0),
+                        &x_y[0],
+                        &x_y[1],
+                    )
+                    .await
+                    .unwrap()
+                })
+                .await
+                .reconstruct();
+
+            assert_eq!(result, <Boolean>::from(expected));
+
+            let result2 = world
+                .semi_honest(records.into_iter(), |ctx, x_y| async move {
+                    compare_geq::<_, BA64, BA64>(
+                        ctx.set_total_records(1),
+                        protocol::RecordId(0),
+                        &x_y[0],
+                        &x_y[0],
```

Comment:
Add a comment here. I originally thought this was a mistake until I realized you were just exercizing the "equals" case.